src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIRLoweringPhase.cs

```diff
@@ -134,12 +134,19 @@ public override void VisitImportSpan(AddImportChunkGenerator chunkGenerator, Spa
 
             public override void VisitAddTagHelperSpan(AddTagHelperChunkGenerator chunkGenerator, Span span)
             {
-                _builder.Push(new DirectiveIRNode()
+                var directiveNode = new DirectiveIRNode()
                 {
                     Name = CSharpCodeParser.AddTagHelperDirectiveDescriptor.Directive,
                     Descriptor = CSharpCodeParser.AddTagHelperDirectiveDescriptor,
                     Source = BuildSourceSpanFromNode(span),
-                });
+                };
+
+                for (var i = 0; i < chunkGenerator.Diagnostics.Count; i++)
+                {
+                    directiveNode.Diagnostics.Add(chunkGenerator.Diagnostics[i]);
+                }
+
+                _builder.Push(directiveNode);
```

Comment:
Why not just create the error node directly here? I'd rather have the fork in lowering than a separate pass